tests/test_pep518.py

```diff
@@ -15,7 +15,6 @@
 @pytest.mark.parametrize(
     'build_arg', ['', '--wheel'], ids=['sdist_to_wheel', 'wheel_directly']
 )
-@pytest.mark.xfail(sys.platform.startswith('cygwin'), reason='ninja pip package not available on cygwin', strict=True)
 def test_pep518(pep518, package, build_arg, tmp_path):
```

Comment:
It would be cool to extend this test to be an integration test for the two cases: ninja from a package and ninja from the system. However I'm not so sure about what strategy we could use to realize that. If we want to keep the test not dependent on a specific test environment setup, the only thing I can think about is to provide a wheelhouse that does not have ninja, mask a possibly existing ninja by adding a script that returns an error or a wrong version number to a directory we prepend to `$PATH`, and verify that the build fails. Add ninja to the wheelhouse and verify that the build succeeds.